### PR TITLE
WT-2572: don't select an in-memory format run if incompatible options configured

### PR DESCRIPTION
**Pull request by keithbostic**
@sueloverso, for your review.

### CODE REVIEW

**Comment by sueloverso** [comment]
@keithbostic That looks good to me for the recovery test.  Feel free to merge when testing completes.

File: test/format/config.c [hunk 1]
```diff
@@ -331,11 +328,43 @@ config_encryption(void)
...
+ *	Periodically set up an in-memory configuration.
  */
 static void
 config_in_memory(void)
 {
+	/*
+	 * Configure in-memory before configuring anything else, in-memory has
+	 * many related requirements. Don't configure in-memory if there's any
+	 * incompatible configurations, so we don't have to configure in-memory
+	 * every time we configure something like LSM, that's too painful.
```
Comment (sueloverso) [inline test/format/config.c]: The comment mentions LSM, but where do we check for LSM?  The others look good.  I think we want to add "compact" to the list.

Comment (sueloverso) [inline test/format/config.c]: Actually, compact seems to be a no-op, not an error.

Comment (keithbostic) [inline test/format/config.c]: @sueloverso, thanks, you're right, and I've added the additional check.

**Comment by sueloverso** [comment]
Thanks! LGTM once testing completes.

**Pull request merged**